railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -141,6 +141,7 @@ def test
     end
 
     def tmp
+      empty_directory_with_keep_file "tmp"
```

Comment:
I don't know. If we remove the creation of the tmp directory in rake restart, wouldn't cloning an older app still cause the original problem?

On the other hand is it the intended behavior to have the tmp folder exist after cloning? If we want it to exist, then we do need the .keep..